app/pulp/app/viewsets/repository.py

```diff
@@ -72,6 +72,15 @@ class ImporterViewSet(NamedModelViewSet):
     serializer_class = ImporterSerializer
     endpoint_name = 'importers'
 
+    def destroy(self, request, name):
+        imp = self.get_object()
+        async_result = importer.delete.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_IMPORTER_TYPE, imp.name,
```

Comment:
The first argument for the `apply_async_with_reservation` is `resource_type` and the second - `resource_id`.
I believe for creating tasks we use only one resource type right now and it is repository and the id is its name. 
The rest of the "resource types" are specified only in tags.

This way we avoid running certain tasks on the same repository simultaneously:
```
importer.delete.apply_async_with_reservation(tags.RESOURCE_REPOSITORY_TYPE, imp.repository.name, ...)
```